peer/main.go

```diff
@@ -392,11 +395,15 @@ func serve(args []string) error {
 		viper.Set("ledger.blockchain.deploy-system-chaincode", "false")
 		viper.Set("validator.validity-period.verification", "false")
 	}
+
+	// Install discovery
+	discovery.SetDiscoveryService(core.NewStaticDiscovery(viper.GetBool("peer.validator.enabled")))
```

Comment:
Why do we use a singleton, which even gets initialized the same way?  I think (explicitly) passing the new discovery instance to the peer is better (than implicitly sharing it via a global singleton).